fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -181,14 +227,15 @@ private void insertField(LuceneDocumentFromRecord.DocumentField field, final Doc
         document.add(luceneField);
     }
 
-    private void writeDocument(@Nonnull List<LuceneDocumentFromRecord.DocumentField> fields, Tuple groupingKey, byte[] primaryKey) throws IOException {
+    private void writeDocument(@Nonnull List<LuceneDocumentFromRecord.DocumentField> fields, Tuple groupingKey,
+                               byte[] primaryKey, @Nullable AnalyzingInfixSuggester suggester) throws IOException {
         final IndexWriter newWriter = getOrCreateIndexWriter(state, indexAnalyzer, executor, groupingKey);
         BytesRef ref = new BytesRef(primaryKey);
         Document document = new Document();
         document.add(new StoredField(PRIMARY_KEY_FIELD_NAME, ref));
         document.add(new SortedDocValuesField(PRIMARY_KEY_SEARCH_NAME, ref));
-        for (LuceneDocumentFromRecord.DocumentField field : fields ) {
-            insertField(field, document);
+        for (LuceneDocumentFromRecord.DocumentField field : fields) {
```

Comment:
Would this add the suggester to the primary key fields? is that what we want?

Reply:
We probably want to ignore the primary key field here since it may contain fields (in my understanding) that the user isn't intending to index/make searchable?

Reply:
The suggester only adds the term in for ```TEXT``` fields. So I assume all fields with ```TEXT``` type are searchable?

Reply:
maybe? probably. Thats more one of those use case conditions thats not written. But its ok for now.